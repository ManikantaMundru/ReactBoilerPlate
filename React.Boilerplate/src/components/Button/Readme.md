
# enabled Button example

```js
<Button title="test" type="submit" disabled={false} onClick={() => alert('clicked button')} />
```

# disabled Button example

```js
<Button title="test" type="submit" disabled={true} onClick={() => alert('clicked disabled')}/>
```
