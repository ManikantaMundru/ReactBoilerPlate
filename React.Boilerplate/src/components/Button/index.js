/**
 *
 * Button
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import StyledButton from './styledComponents/styledButton';

function Button(props) {
  const { title, disabled, onClick, type } = props;

  const handleClick = () => {
    if (typeof onClick === 'function' && onClick) {
      onClick();
    }
  };
  return (
    <StyledButton type={type} disabled={disabled} onClick={handleClick}>
      {title}
    </StyledButton>
  );
}

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
  title: PropTypes.string.isRequired,
  type: PropTypes.string,
};

export default Button;
