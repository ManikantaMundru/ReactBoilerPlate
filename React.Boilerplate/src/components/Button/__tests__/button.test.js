import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Button from '../index';

afterEach(cleanup);

test('does component render', () => {
  const { asFragment } = render(
    <Button title="test" onClick={() => alert('clicked')} />, // eslint-disable-line
  );
  expect(asFragment()).toMatchSnapshot();
});
