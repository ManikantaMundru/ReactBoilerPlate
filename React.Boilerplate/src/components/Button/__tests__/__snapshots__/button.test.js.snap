// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does component render 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa fowqLv"
  >
    test
  </button>
</DocumentFragment>
`;
